Escaping SQLCMD values that contain complex strings

Working on creating an automatic versioning table that uses MSBuild properties injected into SQLCMD variables that are used in a post-deploy script:

```xml
<Project Sdk="MSBuild.Sdk.SqlProj/2.6.0">
    <PropertyGroup>
        <TargetFramework>netstandard2.0</TargetFramework>
        <SqlServerVersion>Sql150</SqlServerVersion>
        <!-- For additional properties that can be set here, please refer to https://github.com/rr-wfm/MSBuild.Sdk.SqlProj#model-properties -->
    </PropertyGroup>

    <PropertyGroup>
        <!-- Refer to https://github.com/rr-wfm/MSBuild.Sdk.SqlProj#publishing-support for supported publishing options -->
    </PropertyGroup>

    <ItemGroup>
      <Content Include="..\Database\**\*.sql"/>
    </ItemGroup>

    <!-- standard Post-Deployment scripts -->
    <ItemGroup>
        <PostDeploy Include="..\Database\Scripts\PostDeploy\PostDeployment.AddVersionInformation.sql" />
    </ItemGroup>

    <!-- SQLCmd variables -->
    <ItemGroup>
    <SqlCmdVariable Include="SchemaVersionNumber">
      <DefaultValue>0.1.0</DefaultValue>
      <Value>$(VersionPrefix)</Value>
    </SqlCmdVariable>
    <SqlCmdVariable Include="SchemaVersionReleaseNotes">
      <DefaultValue>Release notes</DefaultValue>
      <Value>$(PackageReleaseNotes)</Value>
    </SqlCmdVariable>
  </ItemGroup>
</Project>
```

```sql
IF '[$(SchemaVersionNumber)]' = '[]'
BEGIN
    RAISERROR('The version number is an empty string and cannot be inserted.', 16, 1);
    RETURN; -- Exit the script
END
GO
--PRINT N'Attempting to insert the new version number [' + $(SchemaVersionNumber) + N'] into the SchemaChangeLog table.'

IF NOT EXISTS (
    SELECT 1 
    FROM [dbo].[SchemaChangeLog] 
    WHERE [VersionNumber] = N'$(SchemaVersionNumber)'  -- Notice the N'...' around the variable
)
BEGIN
    -- If it doesn't exist, insert the new row.
    INSERT INTO [dbo].[SchemaChangeLog] 
        ([VersionNumber], [TimeDeployed], [ReleaseNotes])
    VALUES
        (N'$(SchemaVersionNumber)', GETDATE(), N'$(SchemaVersionReleaseNotes)')  -- Notice the () instead of []
END
```

Problem I'm running into while running `dotnet build -c Release`:

```
  Unrecognized command or argument 'notes'.

C:\Users\aaron\.nuget\packages\msbuild.sdk.sqlproj\2.6.0\Sdk\Sdk.targets(242,5): error MSB3073: The command "dotnet "C:\Users\aaron\.nuget\packages\msbuild.sd k.sqlproj\2.6.0\Sdk\../tools/net7.0/DacpacTool.dll" build -o "obj\Release\netstandard2.0\Sdkbin.Sql.Build.dacpac" -n "Sdkbin.Sql.Build" -v "0.1.0" -sv Sql150
-i "obj\Release\netstandard2.0\Sdkbin.Sql.Build.InputFiles.txt"  -sc SchemaVersionNumber=0.1.0 -sc SchemaVersionReleaseNotes=Release notes  -dp IncludeComposi teObjects=true  --postdeploy ..\Sdkbin.Database\Scripts\PostDeploy\PostDeployment.AddVersionInformation.sql       " exited with code 1. [E:\Repositories\sdkbi n-2.0\sdkbin-sql\src\Sdkbin.Sql.Build\Sdkbin.Sql.Build.csproj]
```

My `$(PackageReleaseNotes)` is defined in `Directory.Build.props` and contains markdown syntax:

```md
* Initial release

---
code sample
---

More text that belongs in the release notes
```

What can I do to escape the value of this MSBuild variable when it's passed into SQLCMD?

--- a/src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets
+++ b/src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets
@@ -225,7 +225,7 @@
       <InputFileArguments>-i &quot;$(IntermediateOutputPath)$(MSBuildProjectName).InputFiles.txt&quot;</InputFileArguments>
       <BuildPropertyArguments>@(BuildPropertyNames->'-bp %(Identity)=%(PropertyValue)', ' ')</BuildPropertyArguments>
       <DeployPropertyArguments>@(DeployPropertyNames->'-dp %(Identity)=%(PropertyValue)', ' ')</DeployPropertyArguments>
-      <SqlCmdVariableArguments>@(SqlCmdVariable->'-sc %(Identity)=%(DefaultValue)', ' ')</SqlCmdVariableArguments>
+      <SqlCmdVariableArguments>@(SqlCmdVariable->'-sc %(Identity)=&quot;%(DefaultValue)&quot;', ' ')</SqlCmdVariableArguments>
       <PreDeploymentScriptArgument>@(PreDeploy->'--predeploy %(Identity)', ' ')</PreDeploymentScriptArgument>
       <PostDeploymentScriptArgument>@(PostDeploy->'--postdeploy %(Identity)', ' ')</PostDeploymentScriptArgument>
       <RefactorLogScriptArgument>@(RefactorLog->'--refactorlog %(Identity)', ' ')</RefactorLogScriptArgument>

--- a/test/TestProjectWithSqlCmdVariables/TestProjectWithSqlCmdVariables.csproj
+++ b/test/TestProjectWithSqlCmdVariables/TestProjectWithSqlCmdVariables.csproj
@@ -11,7 +11,7 @@
       <Value>$(SqlCmdVar__1)</Value>
     </SqlCmdVariable>
     <SqlCmdVariable Include="MySqlCmdVariable2">
-      <DefaultValue>DefaultValue</DefaultValue>
+      <DefaultValue>Some value with spaces and such</DefaultValue>
       <Value>$(SqlCmdVar__2)</Value>
     </SqlCmdVariable>
   </ItemGroup>

